federatedscope/llm/offsite_tuning/utils.py

```diff
@@ -0,0 +1,115 @@
+import gc
+import copy
+import logging
+import torch
+import torch.nn as nn
+
+from transformers import (
+    OPTForCausalLM,
+    GPT2LMHeadModel,
+    BloomForCausalLM,
+)
+from federatedscope.llm.model.adapter_builder import AdapterModel
+
+logger = logging.getLogger(__name__)
+
+
+def get_layers(adapter_model):
```

Comment:
If these codes are from their official codes, please point out it such as 
`most of the codes are modified from the official implementation`. After we adding more novel implementations, we can remove this.